Add `qml.to_openqasm` transform (#7393)
PennyLaneAI/pennylane: doc/releases/changelog-dev.md

<h3>New features since last release</h3>

> * A new function called `qml.to_openqasm` has been added, which allows for converting PennyLane circuits to OpenQASM 2.0 programs.

Author: Same here @isaacdevlugt

> * A new function called :func:~.to_openqasm has been added, which allows for converting PennyLane circuits to OpenQASM 2.0 programs.

Reviewer: ![image](https://github.com/user-attachments/assets/73d59a95-e7d4-4ca5-9db4-02bb58496685)

> * A new function called :func:~.to_openqasm has been added, which allows for converting PennyLane circuits to OpenQASM 2.0 programs.

Reviewer: ```suggestion
* A new function called :func:`~.to_openqasm` has been added, which allows for converting PennyLane circuits to OpenQASM 2.0 programs.
```
Guess this might help

Author: Thank you @JerryChen97!

<h3>New features since last release</h3>

* The `qml.to_openqasm` function has been implemented in the `io` module to convert a quantum circuit to its representation as an OpenQASM 2.0 program.
  [(#7393)](https://github.com/PennyLaneAI/pennylane/pull/7393)

> 

Author: Ok thank you very much for the suggestion @isaacdevlugt! I just added a code example in the changelog (see [7d2e729](https://github.com/PennyLaneAI/pennylane/pull/7393/commits/7d2e72969fceac1ed6268ca81e7f6432cb9b03a3))

* Two new functions called :func:`~.math.convert_to_su2` and :func:`~.math.convert_to_su4` have been added to `qml.math`, which convert unitary matrices to SU(2) or SU(4), respectively, and optionally a global phase.
  [(#7211)](https://github.com/PennyLaneAI/pennylane/pull/7211)

> * The transform `to_openqasm` is added to the `io` module to convert a `QNode` circuit to his representation as an OpenQASM 2.0 program.
>   [(#7393)](https://github.com/PennyLaneAI/pennylane/pull/7393)

Reviewer 2: ```suggestion
* The `qml.to_openqasm` function has been implemented in the `io` module to convert a quantum circuit to its representation as an OpenQASM 2.0 program.
  [(#7393)](https://github.com/PennyLaneAI/pennylane/pull/7393)
```

> 

Reviewer 2: Feel free to move this on top of the `New features since last release` header (we usually place the most recent entries starting from top)

Author: Oook done, thank you again!
